src/tests/tenderlyKeyHandler.test.ts

```diff
@@ -0,0 +1,85 @@
+import { TenderlySettings } from 'src/types';
+
+import { beforeEach, beforeAll, describe, expect, it, vi } from 'vitest';
+import { tenderlyKeyHandler } from '~/commands/settings/handlers/tenderlyKeyHandler';
+
+import * as keys from '~/lib/keys';
+import * as prompts from '~/lib/prompts.js';
+
+vi.mock('~/lib/keys');
+
+describe('tenderlyKeyHandler', () => {
+  const mockFunction = (() => {}) as any as () => never;
+
+  beforeAll(() => {
+    console.log = vi.fn(); // suppress console.log
+  });
+
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
+  it('should prompt and store a new key if no existing key is found and none provided', async () => {
+    const newSettings: TenderlySettings = {
```

Comment:
Since we're using this same object to test every time, we could store it once in the `mocks` folder and call it when needed or even at the top of the file (if we only use it in this file)